lib/record_store/record/txt.rb

```diff
@@ -2,28 +2,37 @@ module RecordStore
   class Record::TXT < Record
     attr_accessor :txtdata
 
-    validates :txtdata, presence: true, length: { maximum: 255 }
-    validate :escaped_semicolons
+    validates :txtdata, presence: true, length: { maximum: 4096 }
+    validate :escaped_semicolons, :long_txt_quoting
 
     def initialize(record)
       super
-      @txtdata = record.fetch(:txtdata)
+      txt = record.fetch(:txtdata)
+      Record.needs_long_quotes?(txt) ? @txtdata = Record.quote(txt) : @txtdata = txt
     end
 
     def rdata
       { txtdata: txtdata }
     end
 
     def rdata_txt
-      Record.quote(txtdata)
+      txtdata.length > 255 ? txt = txtdata : txt = Record.quote(txtdata)
```

Comment:
Same two as above.
And as you're returning `txt` anyway, assignments are redundant here